modbus_cli/access.py

```diff
@@ -79,7 +79,7 @@ def print_values(self, definitions=None):
         for label, value, presenter in zip(self.labels(), self.values, self.presenters):
             if len(value) == 1:
                 value = value[0]
-            print('{}: {} {}'.format(label, value, self.present_value(value, presenter, definitions)))
+            logging.info('{}: {} {}'.format(label, value, self.present_value(value, presenter, definitions)))
```

Comment:
This is an extreme interpretation of what is considered logging. I prefer to keep it a simple `print()` so that the format of the output is always unambiguously defined just in this line (in particular the consumers of the output won't need to worry about the logging metadata.)

Reply:
Heh, I'm the opposite, I have the habit of _never_ `print()`ing anything but transient debugging information where I intend to remove the print later.

I've run into a lot of contexts where I'm attempting to (re) use non-library python code as a library, and having the output fed into a logger (ideally with a unique logger path, which we admittedly don't have here) means one can prevent external code or similar from outputting to stdout without having to actually hack on it.

In this case, that'd probably not be a major concern, but well, habits.

Reply:
I see your point about using the code as library, in which case I never use printing either. But I don't see it happening soon. If needed we need to rework the code a bit, perhaps introducing callback or similar.

I'm more concerned about putting the burden on the consumer of the output to strip the timestamp, log level etc. before consuming the value they are interested in.

Reply:
None of those are present in the output, as the custom logging handler I added for controlling colorization as a function of log level doesn't attach that. 

From a user perspective, this behaves exactly like it did before without the `-v` flag, and effectively identical _with_ the `-v` flag excepting the difference in how the coloring works.

Reply:
```
durr@shedmon ~> modbus /dev/ttyS5 -b 9600 -p 1 -s 1 h@0/98h
Parsed 0 registers definitions from 1 files
0: (7, 1, 0, 0, 0, 1, 10, 10, 0, 17131, -1311, 0, 0, 17128, -20972, 16602, 15729, 0, 0, 16528, 23069, 16497, -17826, 17542, 24576, 17432, 16384, 16222, 30409, 17007, -12321, 0, 0, 17290, -6554, 17169, -6554, 0, 0, 17438, 16384, 0, 0, 17365, 0, 17315, -32768, 0, 0, 17293, 0, 17461, -32768, 0, 0, 17407, -32768, 17562, 24576, 16228, 23069, 16255, -16777, 16213, 16253, 17131, -2621, 17226, -4588, 17128, -22282, 0, 0, 16862, 15729, 16745, 28836, 0, 0, 0, 1000, 0, 1000, 0, 1000, 0, 1000, 0, 1000, 0, 1000, 0, 1000, 0, 1000, 0, 11520, 1)
```
```
durr@shedmon ~> modbus /dev/ttyS5 -b 9600 -p 1 -s 1 h@0/98h -v
Parsed 0 registers definitions from 1 files
Serial port /dev/ttyS5. Parameters: 9600 baud, 1 stop bit(s), parity: N, timeout 5.0s.
→ < 01 03 00 00 00 62 c4 23 >
← < 01 03 c4 00 07 00 01 00 00 00 00 00 00 00 01 00 0a 00 0a 00 00 42 ec 28 f6 00 00 00 00 42 e9 38 52 40 e1 0e 56 00 00 00 00 40 94 08 31 40 78 b4 39 44 8d c0 00 44 1d 40 00 3f 5f be 77 42 70 17 8d 00 00 00 00 43 8a e6 66 43 11 e6 66 00 00 00 00 44 27 80 00 00 00 00 00 43 de 80 00 43 aa 80 00 00 00 00 00 43 90 00 00 44 40 00 00 00 00 00 00 44 04 80 00 44 a2 20 00 3f 65 1e b8 00 00 00 00 3f 57 0a 3d 42 ec 28 f6 43 4b 3d 71 42 e9 38 52 00 00 00 00 41 de 3d 71 41 69 70 a4 00 00 00 00 00 00 03 e8 00 00 03 e8 00 00 03 e8 00 00 03 e8 00 00 03 e8 00 00 03 e8 00 00 03 e8 00 00 03 e8 00 00 2d 00 00 01 e5 2f > 201 bytes
← [7, 1, 0, 0, 0, 1, 10, 10, 0, 17132, 10486, 0, 0, 17129, 14418, 16609, 3670, 0, 0, 16532, 2097, 16504, 46137, 17549, 49152, 17437, 16384, 16223, 48759, 17008, 6029, 0, 0, 17290, 58982, 17169, 58982, 0, 0, 17447, 32768, 0, 0, 17374, 32768, 17322, 32768, 0, 0, 17296, 0, 17472, 0, 0, 0, 17412, 32768, 17570, 8192, 16229, 7864, 0, 0, 16215, 2621, 17132, 10486, 17227, 15729, 17129, 14418, 0, 0, 16862, 15729, 16745, 28836, 0, 0, 0, 1000, 0, 1000, 0, 1000, 0, 1000, 0, 1000, 0, 1000, 0, 1000, 0, 1000, 0, 11520, 1]
0: (7, 1, 0, 0, 0, 1, 10, 10, 0, 17132, 10486, 0, 0, 17129, 14418, 16609, 3670, 0, 0, 16532, 2097, 16504, -19399, 17549, -16384, 17437, 16384, 16223, -16777, 17008, 6029, 0, 0, 17290, -6554, 17169, -6554, 0, 0, 17447, -32768, 0, 0, 17374, -32768, 17322, -32768, 0, 0, 17296, 0, 17472, 0, 0, 0, 17412, -32768, 17570, 8192, 16229, 7864, 0, 0, 16215, 2621, 17132, 10486, 17227, 15729, 17129, 14418, 0, 0, 16862, 15729, 16745, 28836, 0, 0, 0, 1000, 0, 1000, 0, 1000, 0, 1000, 0, 1000, 0, 1000, 0, 1000, 0, 1000, 0, 11520, 1)
```

Reply:
:+1:

Reply:
I mean, I'm not dead set on this, it was just kind of a "lets move all the print output to the new logger" thing. I'm happy either way.